libs/coin-tester/src/main.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  AccountBridge,
+  Account,
+  TransactionCommon,
+  SignOperationEvent,
+  CurrencyBridge,
+} from "@ledgerhq/types-live";
+import { AssertionError } from "assert";
+import chalk from "chalk";
+import { first, firstValueFrom, map, reduce } from "rxjs";
+
+export type ScenarioTransaction<T> = T & {
+  name: string;
+  expect?: (account: Account) => void;
+};
+
+export type Scenario<T extends TransactionCommon> = {
+  name: string;
+  setup: () => Promise<{
+    accountBridge: AccountBridge<T>;
+    currencyBridge: CurrencyBridge;
+    account: Account;
+    testTimeout?: number;
+    retryInterval?: number;
+    onSignerConfirmation?: (e?: SignOperationEvent) => Awaited<void>;
+  }>;
+  transactions: ScenarioTransaction<Partial<T>>[];
+  beforeAll?: () => Awaited<void>;
+  afterAll?: () => Awaited<void>;
```

Comment:
I think we kinda missed that we're not testing the account before and after the scenario execution. 
`beforeAll` & `afterAll` should have `account` as parameter and the scenario should be able to set some `expect` here based on it